components/menu/sub_menu.tsx

```diff
@@ -0,0 +1,240 @@
+
+    // Menu props
+    menuId: string;
+    menuAriaLabel?: string;
+    forceOpenOnLeft?: boolean; // Most of the times this is not needed, since submenu position is calculated and placed
+
+    children: ReactNode;
+
+}
+
+export function SubMenu({id, leadingElement, labels, trailingElements, isDestructive, menuId, menuAriaLabel, forceOpenOnLeft, children, ...rest}: Props) {
+    const [anchorElement, setAnchorElement] = useState<null | HTMLElement>(null);
+    const isSubMenuOpen = Boolean(anchorElement);
+
+    const isMobileView = useSelector(getIsMobileView);
+
+    const anyModalOpen = useSelector(isAnyModalOpen);
+
+    const dispatch = useDispatch();
+
+    function handleSubMenuOpen(event: MouseEvent<HTMLLIElement>) {
+        event.preventDefault();
+
+        if (isMobileView) {
+            dispatch(openModal<SubMenuModalProps>({
+                modalId: menuId,
+                dialogType: SubMenuModal,
+                dialogProps: {
+                    menuId,
+                    menuAriaLabel,
+                    children,
+                },
+            }));
+        } else {
+            setAnchorElement(event.currentTarget);
+        }
+    }
+
+    function handleSubMenuClose(event: MouseEvent<HTMLLIElement>) {
+        event.preventDefault();
+        setAnchorElement(null);
+    }
+
+    // This handleKeyDown is on the menu item which opens the submenu
+    function handleSubMenuParentItemKeyDown(event: KeyboardEvent<HTMLLIElement>) {
+        if (event.key === 'ArrowRight' || event.key === 'Enter') {
+            event.preventDefault();
+            setAnchorElement(event.currentTarget);
+        }
+    }
+
+    function handleSubMenuKeyDown(event: KeyboardEvent<HTMLUListElement>) {
+        if (event.key === 'ArrowDown' || event.key === 'ArrowUp') {
+            // Stop the event from propagating upwards since that causes navigation to move by 2 items at a time
+            event.stopPropagation();
+        } else if (event.key === 'ArrowLeft' || event.key === 'Escape') {
+            event.preventDefault();
+            setAnchorElement(null);
+        }
+    }
+
+    useEffect(() => {
+        if (anyModalOpen && !isMobileView) {
+            setAnchorElement(null);
+        }
+    }, [anyModalOpen, isMobileView]);
+
+    const originOfAnchorAndTransform = useMemo(() => getOriginOfAnchorAndTransform(forceOpenOnLeft, anchorElement), [anchorElement]);
+
+    const hasSubmenuItems = Boolean(children);
+    if (!hasSubmenuItems) {
+        return null;
+    }
+
+    const passedInTriggerButtonProps = {
+        id,
+        'aria-controls': menuId,
+        'aria-haspopup': true,
+        'aria-expanded': isSubMenuOpen,
+        disableRipple: true,
+        leadingElement,
+        labels,
+        trailingElements,
+        isDestructive,
+        onClick: handleSubMenuOpen,
+    };
+
+    if (isMobileView) {
+        return (<ParentMenuItem {...passedInTriggerButtonProps}/>);
```

Comment:
```suggestion
        return <ParentMenuItem {...passedInTriggerButtonProps}/>;
```

Reply:
throws eslint error without the bracket

Reply:
not for me

Reply:
but its not-blocking, so I'll let that slip